src/workflows-app/SubmissionConfig.test.js

```diff
@@ -285,3 +302,1391 @@ describe('BaseSubmissionConfig gets proxy urls from Leo', () => {
+    within(row1cells[1]).getByText('file_output');
+    within(row1cells[2]).getByText('File');
+    within(row1cells[3]).getByDisplayValue('target_workflow_1_file_output');
+
+    const row2cells = within(rows[2]).queryAllByRole('cell');
+    expect(row2cells.length).toBe(4);
+    expect(row2cells[0].textContent).toBe('target_workflow_1');
+    within(row2cells[1]).getByText('unused_output');
+    within(row2cells[2]).getByText('String');
+    within(row2cells[3]).getByDisplayValue('');
+  });
+});
+
+describe('Records Table updates', () => {
+  beforeAll(() => {
+    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 1000 });
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 });
+  });
+
+  beforeEach(() => {
+    getConfig.mockReturnValue({ wdsUrlRoot, cbasUrlRoot });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => {
+    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', originalOffsetHeight);
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', originalOffsetWidth);
+  });
+
+  it('should repopulate the record selector when the dropdown selection changes', async () => {
+    // ** ARRANGE **
+    const mockRunSetResponse = jest.fn(() => Promise.resolve(runSetResponse));
+    const mockMethodsResponse = jest.fn(() => Promise.resolve(methodsResponse));
+    const mockSearchResponse = jest.fn((_root, _instanceId, recordType) => Promise.resolve(searchResponses[recordType]));
+    const mockTypesResponse = jest.fn(() => Promise.resolve(typesResponse));
+    const mockWdlResponse = jest.fn(() => Promise.resolve('mock wdl response'));
+
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            getForMethod: mockRunSetResponse,
+          },
+          methods: {
+            getById: mockMethodsResponse,
+          },
+        },
+        WorkspaceData: {
+          queryRecords: mockSearchResponse,
+          describeAllRecordTypes: mockTypesResponse,
+        },
+        WorkflowScript: {
+          get: mockWdlResponse,
+        },
+      };
+    });
+
+    // ** ACT **
+    render(
+      h(BaseSubmissionConfig, {
+        methodId: '123',
+        name: 'test-azure-ws-name',
+        namespace: 'test-azure-ws-namespace',
+        workspace: mockAzureWorkspace,
+      })
+    );
+
+    // ** ASSERT **
+    await waitFor(() => {
+      expect(mockRunSetResponse).toHaveBeenCalledTimes(1);
+      expect(mockTypesResponse).toHaveBeenCalledTimes(1);
+      expect(mockMethodsResponse).toHaveBeenCalledTimes(1);
+      expect(mockSearchResponse).toHaveBeenCalledTimes(1);
+      expect(mockWdlResponse).toHaveBeenCalledTimes(1);
+    });
+    const table = await screen.findByRole('table');
+
+    // ** ACT **
+    const dropdown = await screen.findByLabelText('Select a data table');
+    await act(async () => {
+      await selectEvent.select(dropdown, ['BAR']);
```

Comment:
Is `act` necessary here? It seems like `selectEvent` should handle that internally?

Reply:
It doesn't seem to change the outcome of the tests, however it does result in the ugly `An update to SubmissionConfig was not wrapped in act...` console messages

Reply:
Ok. Let's definitely keep it to avoid that warning then.